src/modules/directory.rs

```diff
@@ -117,6 +117,10 @@ pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
         _ => ["".to_string(), "".to_string(), prefix + &dir_string],
     };
 
+    if config.convert_slash {
```

Comment:
We don't actually need to make `path_vec` mutable for this right? We can shadow the old variable, e.g.

```suggestion
    let path_vec = if config.convert_slash { path_vec.iter().for_each(|i| *i = from_slash(i)).collect() } else { path_vec };
```

though this also might just be my dislike for mutable state getting the better of me.

Reply:
Definitely don't have to implement this if you feel the current approach is better--I'm sort of ambivalent about this.

Reply:
Ahh, you're right, I didn't think about that.

Well, let's keep it for now, and then once the benchmarks are online, you can come back and say "I told you so" :D